core/src/program.rs

```diff
@@ -192,28 +192,77 @@ pub struct Program<EC: EvalCache> {
     pub field: FieldPath,
 }
 
+/// The Possible Input Sources, anything that a Nickel program can be created from
+enum Input<T, S> {
```

Comment:
So in the end, you didn't make it public? Or are you going to do that in a separate PR?